main.py

```diff
@@ -0,0 +1,140 @@
+import numpy as np
+import torch
+from torch.utils.data import DataLoader
+from torchvision import transforms
+from tqdm import tqdm
+import os
+from scipy.io.wavfile import write
+
+
+from audio_diffusion_pytorch import DiffusionAE, UNetV0, VDiffusion, VSampler
+from audio_encoders_pytorch import MelE1d, TanhBottleneck
+
+from src.datasets import MusicCapsDataset
+from src.features import PreProcessor
+from src.features.extractor import WaveformExtractor
+from src.utils.data import TorchDataset, ModelCheckpoint
+
+def create_device():
+    # Device selection priority: CUDA > MPS > CPU
+    if torch.cuda.is_available():
+        return torch.device('cuda')
+    # mps is breaking during training    
+    elif torch.backends.mps.is_available():
+        device = torch.device('mps')
```

Comment:
Device não é retornado no MPS, então talvez seja o erro. Mas se não for, eu coloquei um try/catch de exemplo também.

```suggestion
    elif torch.backends.mps.is_available():
        try:
            device = torch.device('mps')
            # Create a single tensor here, then try to move to mps device
            return device
        except Exception as e:
            print(f"MPS test failed with error: {e}. Using CPU instead")
```